indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -463,10 +472,21 @@ public long push() throws IOException
           )
       );
     }
+    Path finalDescriptorPath = new Path(segmentBasePath, "descriptor.json");
+
+    if ("hdfs".equals(outputFS.getScheme()) || "viewfs".equals(outputFS.getScheme())) {
```

Comment:
The proliferation of scheme checks against hdfs/viewfs is, imo, kind of sketchy and confusing. How about doing something like this:

- Add JobHelper.isHdfs that returns `"hdfs".equals(fileSystem.getScheme()) || "viewfs".equals(fileSystem.getScheme())`
- And JobHelper.makeIndexZipPath, and JobHelper.makeDescriptorJsonPath that use JobHelper.isHdfs to make their decisions.
- Remove JobHelper.makeSegmentOutputPath and JobHelper.makeSegmentOutputPathUptoVersionForHdfs, since they wouldn't be needed anymore.

At least then the checks are more contained.